hopp/simulation/technologies/financial/custom_financial_model.py

```diff
@@ -228,9 +252,155 @@ def execute(self, n=0):
+                "capacity",
+                max([1E-6, self.value("batt_annual_discharge_energy")[0]/365.0]),
+            )  # kWh/day
+        else:
+            pf.set_params(
+                "capacity",
+                max([1E-6, self.value("annual_energy_kwh")/365.0]),
+            )  # kWh/day
+
+        pf.set_params("maintenance", {"value": self.o_and_m_cost(), "escalation": gen_inflation})
+
+        # pf.add_fixed_cost(
+        #     name="Fixed O&M Cost",
+        #     usage=1.0,
+        #     unit="$/year",
+        #     cost=self.o_and_m_cost(),
+        #     escalation=gen_inflation,
+        # )
+
+        pf.set_params(
+            "analysis start year", self.value('analysis_start_year'), # no explicit year in single owner,  # Add financial analysis start year
+        )
+        pf.set_params(
+            "operating life", self.value('analysis_period')
+        )
+        pf.set_params(
+            "installation months",
+            self.value('installation_months'),  # Add installation time to yaml default=0
+        )
+        pf.set_params(
+            "installation cost",
+            {
+                "value": 0,
+                "depr type": "Straight line",
+                "depr period": 4,
+                "depreciable": False,
+            },
+        )
+        pf.set_params("demand rampup", 0)
+        pf.set_params("long term utilization", 1)  # TODO should use utilization
+        pf.set_params("credit card fees", 0)
+        pf.set_params(
+            "sales tax", self.value('sales_tax_rate_state')/100.0
+        )
+        pf.set_params("license and permit", {"value": 00, "escalation": gen_inflation})
+        pf.set_params("rent", {"value": 0, "escalation": gen_inflation})
+        # TODO how to handle property tax and insurance for fully offshore?
+        pf.set_params(
+            "property tax and insurance",
+            self.value('property_tax_rate')/100.0 + self.value('insurance_rate')/100.0,
+        )
+        pf.set_params(
+            "admin expense",
+            self.value("admin_expense_percent_of_sales")/100.0,
+        )
+        pf.set_params(
+            "total income tax rate",
+            self.value('federal_tax_rate')/100.0 + self.value('state_tax_rate')/100.0,
+        )
+        pf.set_params(
+            "capital gains tax rate",
+            self.value('capital_gains_tax_rate')/100.0,
+        )
+        pf.set_params("sell undepreciated cap", True)
+        pf.set_params("tax losses monetized", True)
+        pf.set_params("general inflation rate", gen_inflation)
+        pf.set_params(
+            "leverage after tax nominal discount rate",
+            nominal_discount_rate,
+        )
+        
+        pf.set_params(
+            "debt equity ratio of initial financing",
+            (
+                self.value('debt_percent')
+                / (100 - self.value('debt_percent'))
+            ),
+        )  # TODO this may not be put in right
```

Comment:
Maybe we can get @benjaminkee to weigh in here?

Reply:
This looks correct. ProFAST uses debt to equity ratio, so if debt is 60 and equity is 40, then debt to equity is 1.5.

Reply:
Comment removed